src/librustdoc/html/markdown.rs

```diff
@@ -797,11 +797,50 @@ impl LangString {
         Self::parse(string, allow_error_code_check, enable_per_target_ignores, None)
     }
 
+    fn split(string: &str) -> impl Iterator<Item = &str> {
+        string
+            .split(|c| c == ',' || c == ' ' || c == '\t')
+            .map(str::trim)
+            .filter(|token| !token.is_empty())
+    }
+
     fn parse(
         string: &str,
         allow_error_code_check: ErrorCodes,
         enable_per_target_ignores: bool,
         extra: Option<&ExtraInfo<'_, '_>>,
+    ) -> LangString {
+        let new = Self::parse_inner(
+            string,
+            allow_error_code_check,
+            enable_per_target_ignores,
+            extra,
+            false,
+        );
+
+        let old = Self::parse_inner(
+            string,
+            allow_error_code_check,
+            enable_per_target_ignores,
+            extra,
+            true,
+        );
+
+        assert_eq!(
+            new, old,
+            "New token split differs from old split.\nNew: {:?}\nOld: {:?}",
+            new, old
+        );
```

Comment:
This prints the assertion failure twice:

```
thread 'rustc' panicked at 'assertion failed: `(left == right)`
  left: `LangString { original: "{.no_run .example}", should_panic: false, no_run: false, ignore: None, rust: false, test_harness: false, compile_fail: false, error_codes: [], allow_fail: false, edition: None }`,
 right: `LangString { original: "{.no_run .example}", should_panic: false, no_run: true, ignore: None, rust: true, test_harness: false, compile_fail: false, error_codes: [], allow_fail: false, edition: None }`: New token split differs from old split.
New: LangString { original: "{.no_run .example}", should_panic: false, no_run: false, ignore: None, rust: false, test_harness: false, compile_fail: false, error_codes: [], allow_fail: false, edition: None }
Old: LangString { original: "{.no_run .example}", should_panic: false, no_run: true, ignore: None, rust: true, test_harness: false, compile_fail: false, error_codes: [], allow_fail: false, edition: None }', src/librustdoc/html/markdown.rs:829:9
```

Reply:
Fixed

Reply:
Fixed